common/app/services/play-mode-manager.ts

```diff
@@ -0,0 +1,131 @@
+import { PlayMode } from '@project/common';
+
+export interface ConflictResolutionContext {
+    mode: PlayMode;
+    shouldResetPlaybackRate: boolean;
+}
+
+export interface PlayModeChanges {
+    readonly added: Set<PlayMode>;
+    readonly removed: Set<PlayMode>;
+}
+
+export default class PlayModeManager {
+    private _modes: Set<PlayMode>;
+    private readonly _conflicts: readonly [PlayMode, PlayMode][] = [[PlayMode.condensed, PlayMode.fastForward]];
+
+    constructor(initialModes: Set<PlayMode> = new Set([PlayMode.normal])) {
+        this._modes = new Set(initialModes);
+        if (this._modes.size === 0) {
+            this._modes.add(PlayMode.normal);
+        }
+    }
+
+    static getModeChanges(oldModes: Set<PlayMode>, newModes: Set<PlayMode>): PlayModeChanges {
+        const added = new Set<PlayMode>();
+        const removed = new Set<PlayMode>();
+
+        for (const mode of newModes) {
+            if (!oldModes.has(mode)) {
+                added.add(mode);
+            }
+        }
+
+        for (const mode of oldModes) {
+            if (!newModes.has(mode)) {
+                removed.add(mode);
+            }
+        }
+
+        return { added, removed };
+    }
+
+    static normalizePlayModes(modes: Set<PlayMode>, oldModes: Set<PlayMode>): Set<PlayMode> {
```

Comment:
I think you can remove this method and rely solely on the `toggle` method, as this method seems to essentially be duplicating the conflict detecting logic of `toggle`. There is only one place where you're using the `Binding.playModes` setter. Instead of using the setter you can use `togglePlayMode` instead.